topk-js/tests/collections.spec.ts

```diff
@@ -0,0 +1,137 @@
+import { EmbeddingDataType, VectorDistanceMetric } from '../index';
+import { binaryVector, bool, bytes, f32Vector, float, int, semanticIndex, text, u8Vector, vectorIndex } from '../schema';
+import { newProjectContext, ProjectContext } from './setup';
+
+describe('Collections', () => {
+  let ctx: ProjectContext;
+  let collectionsCreated: string[] = [];
+
+  beforeEach(() => {
+    ctx = newProjectContext();
+    collectionsCreated = [];
+  });
+
+  afterEach(async () => {
+    // Only delete collections created within the test
+    for (const collectionName of collectionsCreated) {
+      try {
+        await ctx.client.collections().delete(collectionName);
+      } catch (error) {
+        console.error(`Error deleting collection ${collectionName}:`, error);
+      }
+    }
+  });
+
+  // Helper function to create a collection and track it
+  const createCollection = async (name: string, schema: any) => {
+    const collection = await ctx.client.collections().create(name, schema);
+    collectionsCreated.push(name);
+    return collection;
+  };
+
+  test('create collection', async () => {
+    const schema = {
+      title: text(),
+      title_embedding: f32Vector(1536)
+        .required()
+        .index(vectorIndex({ metric: VectorDistanceMetric.Euclidean })),
```

Comment:
instead of `VectorDistanceMetric.Euclidean`, we should enable literal types, so `{ metric: "euclidean" | "dot_product" | "etc" }` to make this more user-friendly and avoid having to import `VectorDistanceMetric`